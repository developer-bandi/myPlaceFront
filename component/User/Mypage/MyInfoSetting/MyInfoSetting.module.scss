.infoBlock {
  display: flex;
  color: #5f5f5f;
  align-items: center;
  height: 70px;
  @media screen and (max-width: 1201px) {
    width: 90%;
    margin: auto;
  }
}

.subTitle {
  width: 200px;
  height: 70px;
  border-right: 1px solid #e9edef;
  border-bottom: 1px solid #e9edef;
  display: flex;
  align-items: center;
  padding-left: 20px;
  background-color: #f8f9fa;
  @media screen and (max-width: 1201px) {
    width: 30%;
  }
}

.content {
  padding-left: 20px;
  width: 800px;
  height: 70px;
  display: flex;
  align-items: center;
  border-bottom: 1px solid #e9edef;
  @media screen and (max-width: 1201px) {
    width: 70%;
  }
}

.nicknameInput {
  width: 150px;
  height: 40px;
  border: 1px solid #ddefe4;
  border-radius: 5px;
}

.nicknameUpdateButton {
  height: 38px;
  margin-left: 10px;
  border-radius: 5px;
  background-color: #00aaff;
  color: white;
  width: 60px;
}
