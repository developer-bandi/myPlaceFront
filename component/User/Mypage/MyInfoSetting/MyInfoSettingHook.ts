import {useEffect, useRef, useState} from "react";
import {
  axiosGetMyInfo,
  axiosPatchMyNickname,
} from "../../../../lib/commonFn/api";
import {useIsLabtopOrTabletOrMobile} from "../../../../lib/customHook/mediaQuery";

export interface userInfoContent {
  localId: string;
  nickname: string;
  provider: string;
  createdAt: string;
  email: string;
}

const useMyInfoSetting = () => {
  const [userInfo, setUserInfo] = useState<{
    content?: userInfoContent;
    loading: boolean;
    error: boolean;
  }>({loading: true, error: false});
  const nicknameInputRef = useRef<HTMLInputElement>(null);
  const isLabtopOrTabletOrMobile = useIsLabtopOrTabletOrMobile();

  useEffect(() => {
    const asyncWrapFn = async () => {
      try {
        const axiosUserInfo = await axiosGetMyInfo();
        setUserInfo({
          content: axiosUserInfo.data,
          loading: false,
          error: false,
        });
      } catch (error) {
        setUserInfo({loading: false, error: true});
      }
    };
    asyncWrapFn();
  }, []);

  useEffect(() => {
    if (nicknameInputRef.current !== null && userInfo.content) {
      nicknameInputRef.current.value = userInfo.content.nickname;
    }
  }, [userInfo]);

  const changeNickname = async () => {
    try {
      const confirm = window.confirm("수정하시겠습니까?");
      if (nicknameInputRef.current !== null && confirm) {
        await axiosPatchMyNickname(nicknameInputRef.current.value);
      }
      alert("수정되었습니다");
    } catch (error) {
      alert("에러가 발생하였습니다");
    }
  };

  return {userInfo, changeNickname, nicknameInputRef, isLabtopOrTabletOrMobile};
};

export default useMyInfoSetting;
