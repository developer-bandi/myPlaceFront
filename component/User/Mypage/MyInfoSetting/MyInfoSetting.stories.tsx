import { action } from "@storybook/addon-actions";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { useEffect, useRef } from "react";
import { useIsLabtopOrTabletOrMobile } from "../../../../lib/customHook/mediaQuery";
import MyInfoSetting from "./MyInfoSetting";

export default {
  title: "User/Mypage/MyInfoSetting",
  component: MyInfoSetting,
} as ComponentMeta<typeof MyInfoSetting>;

const Template: ComponentStory<typeof MyInfoSetting> = (arg) => {
  const isLabtopOrTabletOrMobile = useIsLabtopOrTabletOrMobile();
  const ref = useRef<HTMLInputElement>(null);
  useEffect(() => {
    if (ref.current !== null && arg.serverData.content?.nickname) {
      ref.current.value = arg.serverData.content.nickname;
    }
  }, []);
  return (
    <MyInfoSetting
      {...arg}
      isLabtopOrTabletOrMobile={isLabtopOrTabletOrMobile}
      nicknameInputRef={ref}
    />
  );
};

const nomarlContent = {
  localId: "placeId",
  nickname: "place",
  provider: "local",
  createdAt: "2022-11-11",
  email: "place@gmail.com",
};

export const isLoading = Template.bind({});
isLoading.args = {
  serverData: { loading: true, error: false },
};
export const isError = Template.bind({});
isError.args = {
  serverData: { loading: false, error: true },
};

export const isNomarl = Template.bind({});
isNomarl.args = {
  serverData: { content: nomarlContent, loading: false, error: false },
  changeNickname: () =>
    new Promise((resolve) => {
      action("changePage")();
      resolve();
    }),
};
