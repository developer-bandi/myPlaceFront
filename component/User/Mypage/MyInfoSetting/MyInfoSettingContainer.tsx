import MyInfoSetting from "./MyInfoSetting";
import useMyInfoSetting from "./MyInfoSettingHook";

const MyInfoSettingContainer = () => {
  const {userInfo, changeNickname, nicknameInputRef, isLabtopOrTabletOrMobile} =
    useMyInfoSetting();
  return (
    <MyInfoSetting
      userInfo={userInfo}
      changeNickname={changeNickname}
      nicknameInputRef={nicknameInputRef}
      isLabtopOrTabletOrMobile={isLabtopOrTabletOrMobile}
    />
  );
};

export default MyInfoSettingContainer;
