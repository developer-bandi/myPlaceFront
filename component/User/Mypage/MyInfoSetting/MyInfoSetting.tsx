import { RefObject } from "react";
import MyPageNavigation from "../Common/navigation/MyPageNavigation";
import styles from "./MyInfoSetting.module.scss";
import mypage from "../../../../lib/styles/mypage.module.scss";
import searchResultLoading from "../../../../public/searchResultLoading.gif";
import Image from "next/image";
import { setDateYearMonthDay } from "../../../../lib/commonFn/date";
import Link from "next/link";
import { userInfoDataState } from "./MyInfoSettingContainer";

interface PostListProps {
  serverData: userInfoDataState;
  changeNickname: () => Promise<void>;
  nicknameInputRef: RefObject<HTMLInputElement>;
  isLabtopOrTabletOrMobile: boolean;
}

const MyInfoSetting = ({
  serverData,
  changeNickname,
  nicknameInputRef,
  isLabtopOrTabletOrMobile,
}: PostListProps) => {
  if (serverData.loading) {
    return (
      <div className={mypage.mainBlock}>
        {isLabtopOrTabletOrMobile ? null : <MyPageNavigation />}
        <div className={mypage.subBlock}>
          <h1 className={mypage.title}>내 정보</h1>
          <div className={mypage.loading}>
            <Image src={searchResultLoading} alt="loading"></Image>
          </div>
        </div>
      </div>
    );
  } else if (serverData.error) {
    return (
      <div className={mypage.mainBlock}>
        {isLabtopOrTabletOrMobile ? null : <MyPageNavigation />}
        <div className={mypage.subBlock}>
          <h1 className={mypage.title}>내 정보</h1>
          <div className={mypage.error}>에러발생</div>
        </div>
      </div>
    );
  } else {
    return (
      <div className={mypage.mainBlock}>
        {isLabtopOrTabletOrMobile ? null : <MyPageNavigation />}
        <div className={mypage.subBlock}>
          <h1 className={mypage.title}>내 정보</h1>
          <div className={styles.infoBlock}>
            <div className={styles.subTitle}>소셜</div>
            <div className={styles.content}>
              {serverData.content !== undefined
                ? { kakao: "카카오", naver: "네이버", local: "연결안됨" }[
                    serverData.content.provider
                  ]
                : null}
            </div>
          </div>
          <div className={styles.infoBlock}>
            <div className={styles.subTitle}>아이디</div>
            <div className={styles.content}>
              {serverData.content?.localId || ""}
            </div>
          </div>
          <div className={styles.infoBlock}>
            <div className={styles.subTitle}>이메일</div>
            <div className={styles.content}>
              {serverData.content !== undefined && serverData.content.email}
            </div>
          </div>
          <div className={styles.infoBlock}>
            <div className={styles.subTitle}>생성일자</div>
            <div className={styles.content}>
              {serverData.content !== undefined &&
                setDateYearMonthDay(serverData.content.createdAt)}
            </div>
          </div>
          <div className={styles.infoBlock}>
            <div className={styles.subTitle}>닉네임</div>
            <div className={styles.content}>
              <input className={styles.nicknameInput} ref={nicknameInputRef} />
              <button
                className={styles.nicknameUpdateButton}
                onClick={() => changeNickname()}
              >
                수정
              </button>
            </div>
          </div>
          <div className={styles.infoBlock}>
            <div className={styles.subTitle}>비밀번호</div>
            <div className={styles.content}>
              {serverData.content?.provider === "local" ? (
                <button className={styles.passwordChangeButton}>
                  <Link href="/user/auth/findpassword">
                    비밀번호 재설정하러 이동하기
                  </Link>
                </button>
              ) : (
                "소셜계정은 비밀번호를 재설정할수 없습니다"
              )}
            </div>
          </div>
        </div>
      </div>
    );
  }
};

export default MyInfoSetting;
