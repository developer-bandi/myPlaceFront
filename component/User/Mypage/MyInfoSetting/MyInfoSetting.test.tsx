import {act, render, renderHook, screen, waitFor} from "@testing-library/react";
import axios from "axios";
import {useRouter} from "next/router";
import * as React from "react";
import newReact from "react";
import MyInfoSetting from "./MyInfoSetting";
import useMyInfoSetting from "./MyInfoSettingHook";

jest.mock("axios");
jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));
const mockRouter = {
  pathname: "",
};
(useRouter as jest.Mock).mockReturnValue(mockRouter);
const mockedAxios = axios as jest.Mocked<typeof axios>;
const mockedUserInfo = {
  localId: "testLocalId",
  nickname: "testNickname",
  provider: "testProvider",
  createdAt: "testCreatedAt",
  email: "testEmail",
};

describe("MyInfoSetting Presentational Test", () => {
  describe("데스크톱 일경우", () => {
    it("로딩중", () => {
      const utils = render(
        <MyInfoSetting
          userInfo={{loading: true, error: false}}
          changeNickname={jest.fn()}
          nicknameInputRef={{current: null}}
          isLabtopOrTabletOrMobile={false}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("loading");
    });
    it("에러발생", () => {
      const utils = render(
        <MyInfoSetting
          userInfo={{loading: false, error: true}}
          changeNickname={jest.fn()}
          nicknameInputRef={{current: null}}
          isLabtopOrTabletOrMobile={false}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("error");
    });
    it("정상 출력", () => {
      const utils = render(
        <MyInfoSetting
          userInfo={{content: mockedUserInfo, loading: false, error: false}}
          changeNickname={jest.fn()}
          nicknameInputRef={{current: null}}
          isLabtopOrTabletOrMobile={false}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("result");
    });
  });
  describe("노트북 이하일 경우", () => {
    it("로딩중", () => {
      const utils = render(
        <MyInfoSetting
          userInfo={{loading: true, error: false}}
          changeNickname={jest.fn()}
          nicknameInputRef={{current: null}}
          isLabtopOrTabletOrMobile={true}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("loading");
    });
    it("에러발생", () => {
      const utils = render(
        <MyInfoSetting
          userInfo={{loading: false, error: true}}
          changeNickname={jest.fn()}
          nicknameInputRef={{current: null}}
          isLabtopOrTabletOrMobile={true}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("error");
    });
    it("정상 출력", () => {
      const utils = render(
        <MyInfoSetting
          userInfo={{content: mockedUserInfo, loading: false, error: false}}
          changeNickname={jest.fn()}
          nicknameInputRef={{current: null}}
          isLabtopOrTabletOrMobile={true}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("result");
    });
  });
});

describe("MyInfoSetting Hook 테스트", () => {
  describe("데이터를 받아오는 useEffect 테스트", () => {
    it("정상적으로 데이터를 받아온 경우", async () => {
      mockedAxios.get.mockImplementation(() =>
        Promise.resolve({
          status: 200,
          data: mockedUserInfo,
        })
      );
      jest.spyOn(newReact, "useRef").mockReturnValue({current: {value: ""}});
      const {result} = renderHook(() => useMyInfoSetting());
      waitFor(() => {
        expect(result.current.userInfo).toStrictEqual({
          content: mockedUserInfo,
          loading: false,
          error: false,
        });
        expect(result.current.nicknameInputRef.current).toBe("testNickname");
      });
    });

    it("에러가 발생한 경우", async () => {
      jest.spyOn(newReact, "useRef").mockReturnValue({current: {value: ""}});
      mockedAxios.get.mockImplementation(() =>
        Promise.reject({
          status: 500,
          data: "error",
        })
      );

      const {result} = renderHook(() => useMyInfoSetting(), {});

      waitFor(() => {
        expect(result.current.userInfo).toStrictEqual({
          loading: false,
          error: true,
        });
      });
    });
  });

  describe("changeNickname 함수 테스트", () => {
    it("정상적으로 변경되는 경우", async () => {
      jest.spyOn(newReact, "useRef").mockReturnValue({current: {value: ""}});
      window.confirm = jest.fn(() => true);
      window.alert = jest.fn();
      mockedAxios.patch.mockImplementation(() =>
        Promise.resolve({
          status: 200,
          data: mockedUserInfo,
        })
      );
      const {result} = renderHook(() => useMyInfoSetting());
      await act(async () => {
        await result.current.changeNickname();
      });
      expect(window.alert).toBeCalledWith("수정되었습니다");
    });
    it("에러가 발생한 경우", async () => {
      window.confirm = jest.fn(() => true);
      window.alert = jest.fn();
      mockedAxios.patch.mockImplementation(() =>
        Promise.reject({
          status: 500,
          data: "data",
        })
      );
      const {result} = renderHook(() => useMyInfoSetting());
      await act(async () => {
        await result.current.changeNickname();
      });
      expect(window.alert).toBeCalledWith("에러가 발생하였습니다");
    });
  });
});
