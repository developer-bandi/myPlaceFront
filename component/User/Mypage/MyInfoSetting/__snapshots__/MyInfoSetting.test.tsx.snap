// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyInfoSetting Presentational Test 노트북 이하일 경우 로딩중 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="loading"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        내 정보
      </h1>
      <div
        class="loading"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="loading"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`MyInfoSetting Presentational Test 노트북 이하일 경우 에러발생 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="error"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        내 정보
      </h1>
      <div
        class="error"
      >
        에러발생
      </div>
    </div>
  </div>
</div>
`;

exports[`MyInfoSetting Presentational Test 노트북 이하일 경우 정상 출력 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="result"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        내 정보
      </h1>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          소셜
        </div>
        <div
          class="content"
        />
      </div>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          아이디
        </div>
        <div
          class="content"
        >
          testLocalId
        </div>
      </div>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          이메일
        </div>
        <div
          class="content"
        >
          testEmail
        </div>
      </div>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          생성일자
        </div>
        <div
          class="content"
        >
          testCreatedAt
        </div>
      </div>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          닉네임
        </div>
        <div
          class="content"
        >
          <input
            class="nicknameInput"
          />
          <button
            class="nicknameUpdateButton"
          >
            수정
          </button>
        </div>
      </div>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          비밀번호
        </div>
        <div
          class="content"
        >
          소셜계정은 비밀번호를 재설정할수 없습니다
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MyInfoSetting Presentational Test 데스크톱 일경우 로딩중 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="loading"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        내 정보
      </h1>
      <div
        class="loading"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="loading"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`MyInfoSetting Presentational Test 데스크톱 일경우 에러발생 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="error"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        내 정보
      </h1>
      <div
        class="error"
      >
        에러발생
      </div>
    </div>
  </div>
</div>
`;

exports[`MyInfoSetting Presentational Test 데스크톱 일경우 정상 출력 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="result"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        내 정보
      </h1>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          소셜
        </div>
        <div
          class="content"
        />
      </div>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          아이디
        </div>
        <div
          class="content"
        >
          testLocalId
        </div>
      </div>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          이메일
        </div>
        <div
          class="content"
        >
          testEmail
        </div>
      </div>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          생성일자
        </div>
        <div
          class="content"
        >
          testCreatedAt
        </div>
      </div>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          닉네임
        </div>
        <div
          class="content"
        >
          <input
            class="nicknameInput"
          />
          <button
            class="nicknameUpdateButton"
          >
            수정
          </button>
        </div>
      </div>
      <div
        class="infoBlock"
      >
        <div
          class="subTitle"
        >
          비밀번호
        </div>
        <div
          class="content"
        >
          소셜계정은 비밀번호를 재설정할수 없습니다
        </div>
      </div>
    </div>
  </div>
</div>
`;
